File: Python3/프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    i = 0
    
    stk = []
    while i < len(arr):
        if not stk:
            stk.append(arr[i])
        elif stk and stk[-1] == arr[i]:
            del stk[-1]
        else:
            stk.append(arr[i])
        
        i += 1
    
    return stk if stk else [-1]

File: Swift/프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.swift
import Foundation

func solution(_ arr:[Int]) -> [Int] {
    var stk = [Int]()
    
    for i in 0..<arr.count {
        if stk.isEmpty {
            stk.append(arr[i])
        } else if stk[stk.count - 1] == arr[i]{
            stk.removeLast()
        } else {
            stk.append(arr[i])
        }
    }
    
    return stk.isEmpty ? [-1] : stk
}